lib/web/apiserver.go

```diff
@@ -282,6 +283,30 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 			return
 		}
 
+		// check proxy and auth node versions (breaking change from v5).
+		// If there is a mismatch with the current proxy client, redirect users to an error page to update proxy.
+		//
+		// DELETE IN 5.0: this block only serves to notify users of v4.3+ to upgrade to v5.0.0
+		if strings.HasPrefix(r.URL.Path, "/web/newuser") {
+			res, err := h.auth.Ping(context.TODO())
+			if err != nil {
+				log.WithError(err).Debugf("Could not ping auth server")
+			} else {
+				proxyVersion := strings.Index(teleport.Version, "4.")
+				authVersion := strings.Index(res.GetServerVersion(), "5.")
+
+				if authVersion == 0 && proxyVersion == 0 {
```

Comment:
these variable names are confusing because they do not represent a version but rather the result of the `strings.Index` method.

```golang
isProxyV4x := strings.Index(teleport.Version, "4.") == 0
isAuthV5x := strings.Index(res.GetServerVersion(), "5.") == 0

if isProxyV4x && isAuthV5x {
 ...
}

```

Reply:
revised